apps/src/StudioApp.js

```diff
@@ -1242,11 +1242,20 @@ StudioApp.prototype.initReadonly = function(options) {
 
 /**
  * Load the editor with blocks.
- * @param {string} blocksXml Text representation of blocks.
+ * @param {string} blocks Text representation of blocks (XML or JSON).
  */
-StudioApp.prototype.loadBlocks = function(blocksXml) {
-  var xml = parseXmlElement(blocksXml);
-  Blockly.Xml.domToBlockSpace(Blockly.mainBlockSpace, xml);
+StudioApp.prototype.loadBlocks = function(blocks) {
+  try {
```

Comment:
try/catch here and below feels odd...can we be explicit about whether we expect blocks to be xml or json?

Reply:
Can we add a condition to check if json/xml? maybe add an isJson helper function?